auth0/resource_auth0_client.go

```diff
@@ -220,9 +219,51 @@ func readClient(d *schema.ResourceData, m interface{}) error {
 	if err != nil {
 		return err
 	}
-	d.SetId(c.ClientID)
+
 	d.Set("client_id", c.ClientID)
 	d.Set("client_secret", c.ClientSecret)
+	d.Set("name", c.Name)
+	d.Set("description", c.Description)
+	d.Set("app_type", c.AppType)
+	d.Set("logo_uri", c.LogoURI)
+	d.Set("is_first_party", c.IsFirstParty)
+	d.Set("oidc_compliant", c.OIDCConformant)
+	d.Set("callbacks", c.Callbacks)
+	d.Set("allowed_logout_urls", c.AllowedLogoutURLs)
+	d.Set("allowed_origins", c.AllowedOrigins)
+	d.Set("web_origins", c.WebOrigins)
+	d.Set("sso", c.SSO)
+	d.Set("sso_disabled", c.SSODisabled)
+	d.Set("cross_origin_auth", c.CrossOriginAuth)
+	d.Set("cross_origin_loc", c.CrossOriginLocation)
+	d.Set("custom_login_page_on", c.CustomLoginPageOn)
+	d.Set("custom_login_page", c.CustomLoginPage)
+	d.Set("custom_login_page_preview", c.CustomLoginPagePreview)
+	d.Set("form_template", c.FormTemplate)
+	d.Set("token_endpoint_auth_method", c.TokenEndpointAuthMethod)
+
+	if c.JWTConfiguration != nil {
+		var jwt_config = make(map[string]string)
+		mapstructure.Decode(c.JWTConfiguration, &jwt_config)
+		d.Set("jwt_configuration", jwt_config)
```

Comment:
Can you revert this one too and remove the `github.com/mitchellh/mapstructure` import? It seemed more straightforward in your previous commit.

Reply:
The previous commit did not work. 

FWIW, `mapstructure` is made by Mitchel Hasimoto of Hashcorp (the makers of Terraform) and is used throughout `github.com/hashicorp/terraform/helper`, so it seems like a good fit.

I can remove it, but I'll need to implement this some other way, since what I had didn't work

Reply:
Alright, fair enough :)